### ISSUE TITLE
Parent-child relationships are ignored for QC categories

### ISSUE THREAD

**Issue by leducvin** [qatrackplus/qatrackplus#644]
Hi,

The admin interface allows us to create QC categories with a parent-child relationship, for example this can be useful if wanting to split a category by beam energy:
![image](https://user-images.githubusercontent.com/22982038/219083083-c4384a9d-89f6-453f-a5e3-916415fc6db0.png)

When adding a new QC category from the admin interface, the relationship is correctly displayed as
```
Dosimetry
---Dosimetry 6MV
...
```

However, when configuring a QC test, the relationship is not displayed at all:
![image](https://user-images.githubusercontent.com/22982038/219084031-17a2f50f-393e-4811-8e2a-fd530dc24c2e.png)

Furthermore, when selecting categories upon performing QC, deselecting a parent category does not deselect tests assigned to a child category, which I would have expected.

For example, if skipping Dosimetry tests, I would also like to skip Dosimetry 6MV, Dosimetry 10 MV, etc.

The parent-child relationship is also not displayed. Ideally, if we would want to fully exploit the parent-child relationship of categories, a collapsible tree could be used?

![image](https://user-images.githubusercontent.com/22982038/219085542-ac3f9b3f-9de0-4b6e-b06a-c875daa53bb6.png)

![image](https://user-images.githubusercontent.com/22982038/219085605-896e0e6d-9f2b-4ee5-a024-471cf42ed33e.png)

**Comment by randlet** [comment]
Yes this would be better if it took the parent / child relationship into account.  If I'm being honest, I wish I didn't introduce category trees as it adds a fair bit of complexity with little gain as it stands.

**Comment by leducvin** [comment]
It's true it's not that much of a gain. I was considering having a very long test list organized in categories and sub-categories, and so a navigable tree of categories would have been somewhat helpful in that case.

I guess even better for me would be to have a tree to navigate list-sublist relationships (with show, hide and skip functionality for sublists).

I'm thinking of a kind of table of contents for the current list you are performing, based on the sublist relationships, if you will. But this is another topic I guess.

**Comment by crcrewso** [comment]
Closed as stale.
